enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4725,52 +5003,49 @@ class AdjointGenerator
     CallInst *const newCall = cast<CallInst>(gutils->getNewFromOriginal(&call));
     IRBuilder<> BuilderZ(newCall);
     BuilderZ.setFastMathFlags(getFast());
-
-    if ((funcName == "cblas_ddot" || funcName == "cblas_sdot") &&
-        called->isDeclaration()) {
-      Type *innerType;
-      std::string dfuncName;
-      if (funcName == "cblas_ddot") {
-        innerType = Type::getDoubleTy(call.getContext());
-        dfuncName = "cblas_daxpy";
-      } else if (funcName == "cblas_sdot") {
-        innerType = Type::getFloatTy(call.getContext());
-        dfuncName = "cblas_saxpy";
-      } else {
-        assert(false && "Unreachable");
-      }
-      Type *castvals[2] = {call.getArgOperand(1)->getType(),
-                           call.getArgOperand(3)->getType()};
+    auto module = gutils->oldFunc->getParent();
+    auto &context = call.getContext();
+    Type *voidType = Type::getVoidTy(context);
+    Type *scalarType = getBLASInnerType(funcName, context);
+    bool handled = false;
+
+    if (funcName == "cblas_ddot" || funcName == "cblas_sdot") {
```

Comment:
This should still only apply if this is a declaration.

Reply:
But you have mentioned last week we should still do the high-level blas handler even if the code has a definition? What is the criteria for handling or not handling here?